doc: add release highlights (#24589)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-3.1.1.rst

This patch release adds the following changes on top of the :ref:`nRF Connect SDK v3.1.0 <ncs_release_notes_3.1.0>`:

> This is the first release of nRF Connect SDK that brings comprehensive :ref:`experimental <software_maturity>` support for the nRF54LM20 DK and nRF54LM20A SoC.

Reviewer: ```suggestion
This is the first release of |NCS| that brings comprehensive :ref:`experimental <software_maturity>` support for the nRF54LM20 DK and the nRF54LM20A SoC.
```

See the `nRF54LM20A System-on-Chip`_ page to learn more.

> This release provides the experimental support for the nRF54LM20A SoC with the following features:

Reviewer: ```suggestion
This release provides experimental support for the nRF54LM20A SoC with the following features:
```

* Protocols:

>   * Bluetooth LE, including Channel Sounding

Reviewer 2: ```suggestion
  * Bluetooth® LE, including Channel Sounding
```

  * 2.4Ghz proprietary

>   * Thread 1.3

Reviewer 3: ```suggestion
  * Thread 1.4
```

  * Matter-over-Thread

> * All nRF54LM20A SoC peripherals, including new USB-HS interface.

Reviewer 4: ```suggestion
* All nRF54LM20A SoC peripherals, including the new USB-HS interface.
```

* DFU with MCUboot enabling dual bank DFU with optional external flash.
* Hardware crypto provided by :ref:`PSA Crypto APIs (hardware accelerated) <ug_nrf54l_cryptography>` for cryptographic operations.
* Google Find My Device and Apple Find My for members of the Apple and Google programs.

> * Out-of-the box support across many standard SDK samples.

Reviewer 4: ```suggestion
* Out-of-the-box support across many standard SDK samples.
```
